app/controllers/workshop_logs_controller.rb

```diff
@@ -110,30 +98,99 @@ def validate_new
+    @workshop_log.media_files.build if @workshop_log.media_files.empty?
+
+    # @sectors = Sector.published.map{ |si| [ si.id, si.name ] }
+    # @files = MediaFile.where(["workshop_log_id = ?", @workshop_log.id])
+
+    @title = params[:windows_type_id] == '3' ? :log_title : :title
+    @agency_title = params[:windows_type_id] == '3' ? :log_title : :name
+
+    @agencies = current_user.projects.uniq
+    agency = params[:agency_id].present? ? Project.where(id: params[:agency_id]).last : @agencies.first
+    @agency_id = agency.id if agency
+  end
+
   def set_default_values
     workshop_log = params[:workshop_log];
-    workshop_log.map do |k, v|
+    workshop_log.to_unsafe_h.map do |k, v|
       if k.include?('_ongoing') || k.include?('_first_time')
         workshop_log[k] = 0 if v.nil? || v.blank?
       end
     end
   end
+
+  def set_index_variables
+    @month_year_options = WorkshopLog.where.not(date: nil)
+                                     .group("DATE_FORMAT(COALESCE(date, created_at), '%Y-%m')")
+                                     .select("DATE_FORMAT(COALESCE(date, created_at), '%Y-%m') AS ym, MAX(COALESCE(date, created_at)) AS max_dt")
+                                     .order("max_dt DESC")
+                                     .map { |record| [Date.strptime(record.ym, "%Y-%m").strftime("%B %Y"), record.ym] }
+    @year_options = WorkshopLog.pluck(Arel.sql("DISTINCT EXTRACT(YEAR FROM COALESCE(date, created_at))"))
+                               .sort
+                               .reverse
+    @facilitators = User.joins(:workshop_logs)
+                        .distinct
+                        .order(:last_name, :first_name)
+    # @workshops = Workshop.joins(:workshop_logs)
+    #                      .order(:title)
+  end
+
+  def set_workshop
+    @workshop = Workshop.find_by(id: params[:workshop_id])
+  end
+
   def log_fields
-    log_fields_params.map do |k, v|
+    log_fields_params.to_unsafe_h.map do |k, v|
       { :form_field_id => k, :answer => v, :report_id => @workshop_log.id }
     end
   end
 
   def log_fields_params
-    params.require(:log_fields)
+    params.fetch(:log_fields, {}).permit!
   end
 
   def ws_params
     params.require(:workshop).permit( sectorable_items: [ :sector_id ] )
   end
 
-  def wsl_params
-    params.require(:workshop_log).permit(:children_ongoing, :children_first_time, :teens_ongoing, :teens_first_time, :adults_ongoing, :adults_first_time, :owner_id, :project_id, :date, :workshop_name, :windows_type_id, :other_description, report_form_field_answers: [:form_field_id, :answer_option_id, :answer]).
+  def workshop_log_params
+    params.require(:workshop_log).permit(
+      :children_ongoing, :children_first_time, :teens_ongoing, :teens_first_time,
+      :adults_ongoing, :adults_first_time, :owner_id, :project_id, :date,
+      :workshop_name, :workshop_id, :windows_type_id, :other_description, #:user,
+      quotable_item_quotes_attributes: [
```

Comment:
nested attributes are here. we're getting Quote records, but still need to confirm we're retaining QuotableItemQuote records. there's a weird delete method in report that will need some more investigation.